### DISCUSSION TITLE
zh/Q-and-A/answer/question10

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#922] [Announcements]
# zh/Q-and-A/answer/question10

怎么用C语言解决这个问题？ 问题描述： C语言中不能用变量来表示元素个数，所以有没有什么方法可以控制数组大小？ image.pngimage.png

https://cygnomatic-controller.github.io/zh/Q-and-A/answer/question10.html

**Comment by 477996850** [comment]
1.可以用动态内存分配（malloc或calloc），用户在运行时可以输入数组的大小，然后malloc 函数动态分配所需的内存。使用完数组后，别忘了用 free() 释放分配的内存。
2.使用宏定义常量，可以通过宏来控制数组大小，便于后续修改和管理。
3.当你使用 malloc 动态分配数组后，如果需要增加或减少数组的大小，可以使用 realloc 函数来调整数组的大小。
